orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/metrics/ZombiePipelineCheckingAgent.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.q.metrics
+
+import com.netflix.spectator.api.BasicTag
+import com.netflix.spectator.api.Registry
+import com.netflix.spectator.api.Tag
+import com.netflix.spinnaker.orca.ExecutionStatus.RUNNING
+import com.netflix.spinnaker.orca.notifications.AbstractPollingNotificationAgent
+import com.netflix.spinnaker.orca.notifications.NotificationClusterLock
+import com.netflix.spinnaker.orca.pipeline.model.Execution
+import com.netflix.spinnaker.orca.pipeline.model.Execution.ExecutionType.ORCHESTRATION
+import com.netflix.spinnaker.orca.pipeline.model.Execution.ExecutionType.PIPELINE
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository
+import com.netflix.spinnaker.orca.q.ExecutionLevel
+import com.netflix.spinnaker.q.metrics.MonitorableQueue
+import net.logstash.logback.argument.StructuredArguments.kv
+import org.slf4j.LoggerFactory
+import org.slf4j.MDC
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression
+import org.springframework.stereotype.Component
+import rx.Scheduler
+import rx.schedulers.Schedulers
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+import java.time.temporal.ChronoUnit.MINUTES
+import java.util.Optional
+
+/**
+ * Monitors a queue and generates Atlas metrics.
+ */
+@Component
+@ConditionalOnExpression(
+  "\${queue.zombie-check.enabled:false}")
+@ConditionalOnBean(MonitorableQueue::class)
+class ZombiePipelineCheckingAgent
```

Comment:
I like this change.

nit: Since we are checking both `PIPELINE` and `ORCHESTRATION`, what about renaming the agent to something like `ZombieExecutionCheckingAgent`?